### PR TITLE
[WIP] Cross Entropy: Adds Sigmoid Cross Entropy With Logits

### PR DESCRIPTION
**Pull request by ManrajGrover**
#### Description
This PR adds [Sigmoid Cross Entropy With Logits](https://www.tensorflow.org/api_docs/python/tf/nn/sigmoid_cross_entropy_with_logits) to the API.

---

##### For repository owners only:

Please remember to apply all applicable tags to your pull request.
Tags: FEATURE, BREAKING, BUG, PERF, DEV, DOC, SECURITY

For more info see: https://github.com/tensorflow/tfjs/blob/master/DEVELOPMENT.md

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-core/1091)

### CODE REVIEW

**Comment by ManrajGrover** [comment]
@dsmilkov @nsthorat This operation would be required to support [tf.losses.sigmoid_cross_entropy](https://www.tensorflow.org/api_docs/python/tf/losses/sigmoid_cross_entropy) in Core API

I'm not sure if gradients need to be added for this operation since it makes use of other operations with gradient already implemented. Could you please have a look and let me know?

**Comment by nsthorat** [comment]
This is looking great! Couple small comments then LGTM.

---

Review status: 0 of 1 LGTMs obtained

---

*[src/ops/sigmoid_cross_entropy.ts, line 3 at r2](https://reviewable.io/reviews/tensorflow/tfjs-core/1091#-LFISXxF9WWNjmnmwvx7:-LFISXxF9WWNjmnmwvx8:b1diex9) ([raw file](https://github.com/tensorflow/tfjs-core/blob/3bc0a203d98f0bcd31c190bd4c613cf802fda6e4/src/ops/sigmoid_cross_entropy.ts#L3)):*
> ```XML
> /**
>  * @license
>  * Copyright 2018 Google Inc. All Rights Reserved.
> ```

Google LLC.

---

*[src/ops/sigmoid_cross_entropy.ts, line 28 at r2](https://reviewable.io/reviews/tensorflow/tfjs-core/1091#-LFIT0U-CbdNSPhJba4k:-LFIT0U-CbdNSPhJba4l:b98g0id) ([raw file](https://github.com/tensorflow/tfjs-core/blob/3bc0a203d98f0bcd31c190bd4c613cf802fda6e4/src/ops/sigmoid_cross_entropy.ts#L28)):*
> ```XML
>    * Computes sigmoid cross entropy given logits.
>    *
>    * Implementation:
> ```

finish this jsdoc

---

*[src/ops/sigmoid_cross_entropy_test.ts, line 3 at r2](https://reviewable.io/reviews/tensorflow/tfjs-core/1091#-LFISZ7zC6q27LIjah6d:-LFISZ7zC6q27LIjah6e:b5nriws) ([raw file](https://github.com/tensorflow/tfjs-core/blob/3bc0a203d98f0bcd31c190bd4c613cf802fda6e4/src/ops/sigmoid_cross_entropy_test.ts#L3)):*
> ```XML
> /**
>  * @license
>  * Copyright 2017 Google Inc. All Rights Reserved.
> ```

2018 Google LLC.

---


*Comments from [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-core/1091#-:-LFITK4N0GcCdnLwh5TY:bbg7ykg)*

**Comment by ManrajGrover** [comment]
@nsthorat Made the required changes. For this operation, should we add gradients?

**Comment by nsthorat** [comment]
No need for gradients because it's a combination of other ops which have gradients defined!

**Comment by nsthorat** [comment]
:lgtm_strong:

---

Review status: :shipit: complete! 1 of 1 LGTMs obtained

---



*Comments from [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-core/1091#-:-LFJ4eZc0nshulZMiqpd:b-jbsf8z)*
